tests/e2e/specFiles/Transactions/receive.spec.ts

```diff
@@ -0,0 +1,135 @@
+import { expect } from 'detox';
+
+import { isBeta, WALLETS_WITH_COINS } from '../../helpers';
+import app from '../../pageObjects';
+
+const DATA_FOR_TRANSACTIONS = {
+  DEFAULT_VALUE: 'Amount',
+  AMOUNT_TO_RECEIVE: '10',
+};
+
+describe('Transactions', () => {
+  beforeEach(async () => {
+    isBeta() && (await app.onboarding.betaVersionScreen.close());
+    await app.developerRoom.tapOnSkipOnboardingButton();
+    await app.navigationBar.changeTab('wallets');
+  });
+
+  describe('Receive', () => {
+    describe('3-Key Vault', () => {
+      describe('@android @iOS @smoke', () => {
+        it('should be possible to see QRCode, wallet address and receive amount for 3-Key Vault wallet', async () => {
+          await app.wallets.createWallet({
+            type: '3-Key Vault',
+            name: '3-Key wallet',
+            fastPublicKey: WALLETS_WITH_COINS['3-Keys Vault'].FAST_KEY,
+            cancelPublicKey: WALLETS_WITH_COINS['3-Keys Vault'].CANCEL_KEY,
+          });
+          await app.wallets.dashboardScreen.tapOnReceiveButton();
+          await expect(app.transactionsReceive.receiveCoinsMainScreen.qrCodeIcon).toBeVisible();
+          await expect(app.transactionsReceive.receiveCoinsMainScreen.walletAdressText).toBeVisible();
+          await expect(app.transactionsReceive.receiveCoinsMainScreen.receiveAmountText).toHaveText(
+            DATA_FOR_TRANSACTIONS.DEFAULT_VALUE,
+          );
+          await app.transactionsReceive.receiveCoinsMainScreen.typeAmountToReceive(
+            DATA_FOR_TRANSACTIONS.AMOUNT_TO_RECEIVE,
+          );
+          await expect(app.transactionsReceive.receiveCoinsMainScreen.receiveAmountText).toHaveText(
+            DATA_FOR_TRANSACTIONS.AMOUNT_TO_RECEIVE,
+          );
+        });
```

Comment:
What about testing other functionalities in this view? Like setting up an amount to share with the address? Or switching between wallets? Or using "Copy" and "Share wallet address"?

Reply:
I agree but I think we can add that later in the next iteration of creating a test framework 🙂